pkg/policy/policies.go

```diff
@@ -44,127 +81,83 @@ type Policies struct {
 	containerFiltersEnabled   uint64 // bitmap of policies that have at least one container filter type enabled
 }
 
-func NewPolicies() *Policies {
+func NewPolicies(cfg PoliciesConfig) *Policies {
 	return &Policies{
 		rwmu:                      sync.RWMutex{},
+		config:                    cfg,
 		version:                   0,
 		bpfInnerMaps:              map[string]*bpf.BPFMapLow{},
-		policiesArray:             [MaxPolicies]*Policy{},
+		policiesArray:             [PolicyMax]*Policy{},
 		filterEnabledPoliciesMap:  map[*Policy]int{},
+		evtsStates:                newEventsStates(),
 		filterUserlandPoliciesMap: map[*Policy]int{},
 		uidFilterMin:              filters.MinNotSetUInt,
 		uidFilterMax:              filters.MaxNotSetUInt,
 		pidFilterMin:              filters.MinNotSetUInt,
 		pidFilterMax:              filters.MaxNotSetUInt,
 		uidFilterableInUserland:   false,
 		pidFilterableInUserland:   false,
-		filterableInUserland:      0,
-		containerFiltersEnabled:   0,
+		filterableInUserland:      PolicyNone,
+		containerFiltersEnabled:   PolicyNone,
 	}
 }
 
-func (ps *Policies) Count() int {
+// EventsStates returns the events states of Policies.
+func (ps *Policies) EventsStates() events.EventsStates {
 	ps.rwmu.RLock()
 	defer ps.rwmu.RUnlock()
 
-	return len(ps.filterEnabledPoliciesMap)
+	return ps.evtsStates
 }
 
-func (ps *Policies) UIDFilterMin() uint64 {
-	return atomic.LoadUint64(&ps.uidFilterMin)
-}
+// Count returns the number of enabled policies.
+func (ps *Policies) Count() int {
+	ps.rwmu.RLock()
+	defer ps.rwmu.RUnlock()
 
-func (ps *Policies) UIDFilterMax() uint64 {
-	return atomic.LoadUint64(&ps.uidFilterMax)
+	return len(ps.filterEnabledPoliciesMap)
 }
 
-func (ps *Policies) PIDFilterMin() uint64 {
-	return atomic.LoadUint64(&ps.pidFilterMin)
-}
+func (ps *Policies) Version() uint16 {
+	ps.rwmu.RLock()
+	defer ps.rwmu.RUnlock()
 
-func (ps *Policies) PIDFilterMax() uint64 {
-	return atomic.LoadUint64(&ps.pidFilterMax)
+	return uint16(ps.version)
 }
 
-func (ps *Policies) SetVersion(version uint16) {
-	atomic.StoreUint32(&ps.version, uint32(version))
-}
+// ContainerFilterEnabled returns a bitmap of policies representing the
+// container filter types enabled.
+func (ps *Policies) ContainerFilterEnabled() uint64 {
+	ps.rwmu.RLock()
+	defer ps.rwmu.RUnlock()
 
-func (ps *Policies) Version() uint16 {
-	return uint16(atomic.LoadUint32(&ps.version))
+	return ps.containerFiltersEnabled
 }
 
-// ContainerFilterEnabled returns a bitmap of policies that have at least
-// one container filter type enabled.
-func (ps *Policies) ContainerFilterEnabled() uint64 {
-	return atomic.LoadUint64(&ps.containerFiltersEnabled)
+// HasContainerFilterEnabled returns true if at least one policy has a
+// container filter type enabled.
+func (ps *Policies) HasContainerFilterEnabled() bool {
```

Comment:
`Policies` encompass the filters, so I assumed that `HasContainerFilterEnabled` would be handy. Perhaps it's a bit confusing since `Policy` has a method called `ContainerFilterEnabled` that returns a bool but `Policies` one (same naming) returns its bitmask. I think changing them name would help here.

Reply:
I changed. Check if it's better now.